src/main/java/com/ericsson/ei/mongo/MongoDBHandler.java

```diff
@@ -363,42 +361,62 @@ private boolean doDrop(String dataBaseName, String collectionName, MongoQuery qu
 
     }
 
-    private MongoCollection<Document> getMongoCollection(String dataBaseName,
+    private MongoCollection<Document> getMongoCollection(String databaseName,
             String collectionName) {
-        if (mongoClient == null)
+        if (mongoClient == null) {
             return null;
-        MongoDatabase db;
+        }
+
+        try {
+            List<String> collectionList = getCollectionList(databaseName);
+            if (!collectionList.contains(collectionName)) {
+                createCollection(databaseName, collectionName);
+            }
+        } catch (MongoClientException e) {
+            LOGGER.error("Failure when handling Mongo collection: {} ", e.getMessage(), e);
+            return null;
+        }
```

Comment:
Maybe make the try catch first and last thing in this method. The code below could be moved into the try catch as well. If you can, please make methods out of the things inside the try as well.

Reply:
Somewhat done in next commit.